packages/framework/src/Component/Translation/ConstraintViolationExtractor.php

```diff
@@ -123,9 +123,10 @@ protected function isParameterExecutionContextInterfaceSubclass(Node\Param $para
      */
     protected function isAddViolationMethodCall(Node $node): bool
     {
+        /** @var \PhpParser\Node\Expr\MethodCall $node */
         return $node->var instanceof Variable
             && in_array($node->var->name, $this->currentExecutionContextVariableNames, true)
-            && $node->name === 'addViolation';
+            && $node->name->name === 'addViolation';
```

Comment:
???

Reply:
In version 3, `$node->name` returns `string`. 

Since version 4,`Identifier` object is returned and to get the string I need to read property name of the `Identifier` object or retype to string (`(string)$node->name`).

Reply:
As example: 
If I parse `$context->addViolation('message')` then
`$node->name->name` returns `addViolation` and `$node->var->name` returns `context`.

That's why it's like this.